content/post/practical-stegen.Rmd

```diff
@@ -0,0 +1,825 @@
+
+
+```{r stegen_t_test}
+t.test(stegen$age ~ stegen$ill, var.equal = TRUE)
+```
+
+The p-value of `r round(t.test(stegen$age ~ stegen$ill, var.equal = TRUE)[["p.value"]], 3)` confirms the previous graphics: the illness does not seem to be linked to age.
+
+
+
+### Is illness linked to gender?
+
+To test the association between gender and illness (2 categorical variables), we
+first build a 2-by-2 (contingency) table, using:
+
+```{r stegen_2by2}
+tab_ill_sex <- table(stegen$ill, stegen$sex)
+tab_ill_sex
+```
+
+Note that proportions can be obtained using `prop.table`:
+```{r stegen_prop_table}
+## basic proportions
+prop.table(tab_ill_sex)
+
+## expressed in %, rounded:
+round(100 * prop.table(tab_ill_sex))
+```
+
+Once a contingency table has been built, the Chi-square test can be run using `chisq.test`:
+
+```{r stegen_chisq}
+chisq.test(tab_ill_sex)
+```
+
+Here, the p-value of `r round(chisq.test(tab_ill_sex)[["p.value"]], 3)` suggests illness is not related to sex either.
+
+
+## Making multiple tests
+
+The natural next step is to run multiple tests for all potential risk factors
+recorded. There are many ways to go about this in **R**, but we will illustrate
+one of the most common workflows:
+
+1. isolate a subset of variables to test (the ones indicative of food consumption) (using `[]`)
+2. for each:
+    i) build contingency table with illness status (using `table`)
+    ii) run a Chi-square test on this table (using `chisq.test`)
+
+All of these have been seen so far; the only missing piece is step 2), which we
+will cover using a very handy function called `lapply`.
+
+
+### Isolating the variables to test
+
+We use the natural structure of the dataset, as variables representing food
+consumption are all the last columns:
+
+```{r stegen_isolate_food}
+names(stegen)
+```
+
+In this case, we need to retain columns 6 to 21, excluding `tportion` and
+`mportion`, which are not binary; this can be done using the subsetting operator
+`[...]`, where `...` are numbers indicating the columns to retain:
+
+```{r stegen_food}
+to_keep <- c(6, 8, 9, 10, 12:21)
+to_keep
+food <- stegen[to_keep]
+food
+```
+
+
+### Building several contingency tables
+
+There are many variables to test, and having to enter separate command lines for
+each would be cumbersome and prone to errors. As a workaround, the function
+`lapply` can be used. This function allows to repeat an operation using a vector
+of inputs, using each element in turn. Its general syntax is
+`lapply(vector_of_inputs, function_to_use, other_arguments)`, where
+`other_arguments` can be any secondary arguments taken by
+`function_to_use`. Here, we use it to build a separate contingency table of each
+variable in `food` crossed with illness status (`stegen$ill`); note that we show
+only the first few tables:
+
+```{r stegen_food_tables, echo = c(1,3), eval = -2}
```

Comment:
This chunk is a bit confusing. You want to echo lines 1 and 3 and evaluate lines 1 and 3... leaving line two to basically be padding

Reply:
If you wanted to only show the first six items without "head", then you would want echo = 1:2